common/views/components/IIIFViewer/parts/MainViewer.js

```diff
@@ -0,0 +1,244 @@
+            <ThumbnailWrapper imageLoaded={thumbLoaded}>
+              <IIIFResponsiveImage
+                width={currentCanvas.width}
+                height={currentCanvas.height}
+                src={urlTemplateThumbnail({
+                  size: `${
+                    smallestWidthImageDimensions
+                      ? smallestWidthImageDimensions.width
+                      : '!100'
+                  },`,
+                })}
+                srcSet={''}
+                sizes={''}
+                alt={''}
+                isLazy={false}
+                lang={null}
+                onLoadHandler={() => {
+                  setThumbLoaded(true);
+                }}
+              />
+            </ThumbnailWrapper>
+          )}
+          {(imageType === 'main' || mainLoaded) && urlTemplateMain && (
+            <ImageViewer
+              id="item-page"
+              infoUrl={infoUrl}
+              width={currentCanvas.width}
+              height={currentCanvas.height}
+              alt={ocrText}
+              urlTemplate={urlTemplateMain}
+              setShowZoomed={setShowZoomed}
+              setZoomInfoUrl={setZoomInfoUrl}
+              rotation={rotation}
+              mainViewerRef={
+                mainViewerRef && mainViewerRef.current
+                  ? mainViewerRef.current
+                  : null
+              }
+              setActiveIndex={setActiveIndex}
+              index={index}
+              onLoadHandler={() => {
+                setMainLoaded(true);
+                setIsLoading(false);
+              }}
+            />
+          )}
+        </>
+      )}
+    </div>
+  );
+}, areEqual);
+
+type Props = {|
+  listHeight: number,
+  mainViewerRef: any,
+  setActiveIndex: number => void,
+  pageWidth: number,
+  canvases: [],
+  canvasIndex: number,
+  setShowZoomed: boolean => void,
+  setZoomInfoUrl: string => void,
+  rotatedImages: [],
+  setShowControls: boolean => void,
+  setIsLoading: boolean => void,
+|};
+
+const MainViewer = ({
+  listHeight,
+  mainViewerRef,
+  setActiveIndex,
+  pageWidth,
+  canvases,
+  canvasIndex,
+  setShowZoomed,
+  setZoomInfoUrl,
+  rotatedImages,
+  setShowControls,
+  setIsLoading,
+}: Props) => {
+  const [isProgrammaticScroll, setIsProgrammaticScroll] = useState(false);
+  const [newScrollOffset, setNewScrollOffset] = useState(0);
+  const [firstRender, setFirstRender] = useState(true);
+  const firstRenderRef = useRef(firstRender);
+  firstRenderRef.current = firstRender;
+  const scrollVelocity = useScrollVelocity(newScrollOffset);
+  const debounceHandleOnItemsRendered = useRef(
+    debounce(handleOnItemsRendered, 500)
+  );
+  const itemHeight = pageWidth * 0.8;
+  let timer;
+  function handleOnScroll({ scrollOffset, scrollUpdateWasRequested }) {
+    setNewScrollOffset(scrollOffset);
+    setIsProgrammaticScroll(scrollUpdateWasRequested);
+    setShowControls(false);
+    if (timer !== null) {
```

Comment:
I'm seeing some funky flashing that seems to be timer related

![flashy](https://user-images.githubusercontent.com/1394592/71989665-f7c2ad00-3229-11ea-9c6a-b0ec81655ccd.gif)

Reply:
Making the timer a `ref` solves it

```
const timer = useRef(null);
  const itemHeight = pageWidth * 0.8;
  function handleOnScroll({ scrollOffset, scrollUpdateWasRequested }) {
    clearTimeout(timer.current);
    setShowControls(false);
    setNewScrollOffset(scrollOffset);
    setIsProgrammaticScroll(scrollUpdateWasRequested);

    timer.current = setTimeout(() => {
      setShowControls(true);
    }, 500);
  }
```